docs/new-agent.md

```diff
@@ -0,0 +1,65 @@
+# Importing a New Agent into the Platform from a Code Base
+
+## Pre-requisites
+
+When deploying a new agent, you may either deploy from source code or from a pre-existing container image. When deploying from source code, Kagenti will retrieve the source code from github, Kagenti will build your agent by deploying the code into a container based up on the Dockerfile you provide. When deploying from an image, it is expected that the agent code already exists in the container image, so the github retrieval of the code and its installation will be skipped.
+
+### Deploying from Source
+Before importing a new agent from source, ensure that:
+1. The agent code is hosted on GitHub and is accessible using the GitHub credentials provided [during the Kagenti installation](https://github.com/kagenti/kagenti/blob/main/docs/demos.md). 
+2. The agent code is organized within a sub-directory of the Git repository (not in the root directory).
+3. The root of the subdirectory contains a Dockerfile.
+
+### Deploying from an Image
+Before importing a new agent from an existing Docker image, ensure that the Docker image is available in an accessible container registry.
+
+## Agent Examples
+
+See the [Kagenti agent examples repo](https://github.com/kagenti/agent-examples) for a variety of agents and MCP tool examples.
+
+## Steps to Import a New Agent
+To import a new agent into the platform, follow these steps:
+
+### Step 1: Access the Import New Agent Section
+- Log in to the Kagenti UI.
+- Navigate to the "Import New Agent" section.
+
+### Step 2: Select the Namespace
+- Choose the namespace where you want to deploy the agent.
+
+### Step 3: Configure Environment Variables
+- Manually add environment variables required by your agent.
+- Alternatively, import environment variables from a `.env` file hosted on GitHub.
+
+### Step 4: Select Deployment Method
+
+#### Deploy from an existing Docker image.
+- Select "deploy from existing image" as the deployment method, and provide the address of the image in a remote container registry
+
+#### Deploy from source code.
+
+1. Select "Build from source" as the deployment method
+2. In "Agent Source Repository URL", enter the root of your github repository where your agent project lives.
```

Comment:
```suggestion
2. In "Agent Source Repository URL", enter the root of your GitHub repository where your agent project lives.
```

Also, you are using a mix of lines ending with a `.` and lines ending without punctuation.  All lines should match -- probably all should end in `.`.